cmd/terramate/cli/cli.go

```diff
@@ -1328,23 +1329,76 @@ func (c *cli) printStacks() {
 	if err != nil {
 		c.fatal("Unable to list stacks", err)
 	}
-
 	c.gitFileSafeguards(false)
 
-	for _, entry := range c.filterStacks(report.Stacks) {
-		stack := entry.Stack
+	filteredStacks := c.filterStacks(report.Stacks)
+
+	if c.parsedArgs.List.RunOrder {
+		c.printListRunOrder(filteredStacks, c.parsedArgs.List.Why)
```

Comment:
The list algorithm should be:

```
status := parseStatusFilter(c.parsedArgs.List.ExperimentalStatus)
report, _ := c.listStacks(mgr, c.parsedArgs.Changed, status)
filteredStacks := c.filterStacks(report.Stacks)

// convert into config.List
stacks := make(config.List[*config.SortableStack], len(entries))
for i, e := range entries {
    stacks[i] = e.Stack.Sortable()
}

// sort
orderedStacks, _, _ := run.Sort(c.cfg(), selectedStacks)

// print as before but ranging over config.List
```

The `list` should not call `c.computeSelectedStacks()` because this pulls more stacks if wants/wanted_by is used.

The `debug show order` command does not need to be changed.